packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1861,6 +1882,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             type: envelope.contents.type,
         };
 
+        // Only collect signal telemetry for messages sent by the current client.
+        if (message.clientId === this.clientId) {
+            if (envelope.metadata !== undefined) {
+                if (envelope.metadata.sequenceNumber !== this._perfSignalData.signalSequenceNumber) {
+                    this._perfSignalData.signalsLost++;
+                }
+                if (this._perfSignalData.signalSequenceNumber % 100 === 1) {
+                    this._perfSignalData.totalElapsedTime += (Date.now() - envelope.metadata.timestamp);
```

Comment:
whenever we receive signal (we are tracking), we should log its time to telemetry. Accumulating it will result in capability to report only averages, which screws distribution. I.e. if we assume that time has normal distribution, and we report an average of 10 signals, then P95 of that sum will be way off, as standard deviation of averages will be 10 smaller than standard deviation of initial random variable (single measurement).

Reply:
I will turn into its own event and remove the report of avg.